pgrx/src/nullable.rs

```diff
@@ -0,0 +1,466 @@
+            Nullable::Null => None,
+        }
+    }
+    #[inline]
+    pub fn as_option(&self) -> Option<&T> {
+        match self {
+            Nullable::Valid(val) => Some(val),
+            Nullable::Null => None,
+        }
+    }
+    #[inline]
+    pub fn as_option_mut(&mut self) -> Option<&mut T> {
+        match self {
+            Nullable::Valid(val) => Some(val),
+            Nullable::Null => None,
+        }
+    }
+    // Returns true if this is a valid value (non-null).
+    #[inline]
+    pub fn is_valid(&self) -> bool {
+        match self {
+            Nullable::Valid(_) => true,
+            Nullable::Null => false,
+        }
+    }
+    // Returns true if this is Null.
+    #[inline]
+    pub fn is_null(&self) -> bool {
+        !self.is_valid()
+    }
+    /// Return `value` if this enum is `Valid(value)`, otherwise panics.
+    #[inline]
+    pub fn unwrap(self) -> T {
+        self.into_option().unwrap()
+    }
+    /// Return `value` if this enum is `Valid(value)`, otherwise panics with
+    /// the error message `msg`.
+    #[inline]
+    pub fn expect<'a, A>(self, msg: &'a A) -> T
+    where
+        &'a A: Into<Cow<'a, str>>,
+    {
+        self.into_option().expect(msg.into().as_ref())
+    }
+    /// Convert to a result, returning `err` if this enum is `Null`.
+    #[inline]
+    pub fn valid_or<E>(self, err: E) -> Result<T, E> {
+        self.into_option().ok_or(err)
+    }
+    /// Maps the `T` value inside a `Nullable<T>::Valid` into the U of
+    /// a `Nullable<U>::Valid` using the provided `func` argument,
+    /// or returns `Null` if this enum is `Null`.
+    #[inline]
+    pub fn map<U, F>(self, func: F) -> Nullable<U>
+    where
+        F: FnOnce(T) -> U,
+    {
+        match self {
+            Nullable::Valid(v) => Nullable::Valid(func(v)),
+            Nullable::Null => Nullable::Null,
+        }
+    }
+    /// Returns `Null` if this value is `Null`, or, if this enum is
+    /// `Valid(value)`, it calls `func(value)` and returns `func`'s result.
+    /// This is very similar to [`Nullable::map()`]. However, `map()`
+    /// maps the `value` inside a `Valid(value)` to another `Valid(value)`,
+    /// while `and_then()` can be used to reject a valid value and return
+    /// null anyway.
+    #[inline]
+    pub fn and_then<U, F>(self, func: F) -> Nullable<U>
+    where
+        F: FnOnce(T) -> Nullable<U>,
+    {
+        match self {
+            Nullable::Valid(v) => func(v),
+            Nullable::Null => Nullable::Null,
+        }
+    }
+}
+
+impl<T> Into<Option<T>> for Nullable<T> {
+    #[inline]
+    fn into(self) -> Option<T> {
+        match self {
+            Nullable::Valid(val) => Some(val),
+            Nullable::Null => None,
+        }
+    }
+}
+impl<T> Into<Nullable<T>> for Option<T> {
+    #[inline]
+    fn into(self) -> Nullable<T> {
+        match self {
+            Some(val) => Nullable::Valid(val),
+            None => Nullable::Null,
+        }
+    }
+}
```

Comment:
Wait, I thought we had a pair of From implementations here?

Reply:
Fixed in f503f442ed9d88d2d51070db320a41d3b21e3a8d